pymeasure/instruments/keysight/keysightN7744C.py

```diff
@@ -0,0 +1,158 @@
+        validator=strict_range,
+        values=[-30, 10]
+    )
+
+    auto_range = Channel.control(
+        "SENSe{ch}:POWer:RANGe:AUTO?", "SENSe{ch}:POWer:RANGe:AUTO %g",
+        """Control whether automatic power ranging is being used by the channel.""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={False: 0, True: 1}
+    )
+
+    averaging_time = Channel.control(
+        "SENSe{ch}:POWer:ATIMe?", "SENSe{ch}:POWer:ATIMe %g",
+        """Control the averaging time of the channel.""",
+    )
+
+    auto_gain = Channel.control(
+        "SENSe{ch}:POWer:GAIN:AUTO?", "SENSe{ch}:POWer:GAIN:AUTO %g",
+        """Control the Auto Gain of the channel.""",
+        validator=strict_discrete_set,
+        map_values=True,
+        values={False: 0, True: 1}
+    )
+
+    zero = Channel.control(
+        "SENSe{ch}:CORRection:COLLect:ZERO?", "SENSe{ch}:CORRection:COLLect:ZERO",
+        """Control the electrical offsets of the channel.""",
+    )
+
+    state = Channel.control(
+        "SENSe{ch}:FUNCtion:STATe?", "SENSe{ch}:FUNCtion:STATe %s",
+        """Control the logging, MinMax, or stability data acquisition function mode.""",
+        validator=strict_discrete_set,
+        values=['LOGGing', 'LOGG', 'STABility', 'STAB', 'MINMax', 'MINM', 'STOP', 'START']
+    )
+
+    logging_parameters = Channel.control(
+        "SENSe{ch}:FUNCtion:PARameter:LOGGing?",
+        "SENSe{ch}:FUNCtion:PARameter:LOGGing %g,%gS",
+        """Control the number of data points and the averaging time for the logging data
+        acquisition function."""
+        # get_process=lambda v: tuple(float(x) for x in v.split(','))
+    )
+
+
+class KeysightN7744C(SCPIMixin, Instrument):
+    """
+    This represents the Keysight N7744C Optical Multiport Power Meter interface.
+
+    .. code-block:: python
+
+        mppm = KeysightN7744C(address)
+
+    """
+
+    def __init__(self, adapter, name="N7744C Optical Multiport Power Meter", **kwargs):
+        super().__init__(
+            adapter, name, **kwargs)
+
+    channel_1 = Instrument.ChannelCreator(MPPMChannel, 1)
+    channel_2 = Instrument.ChannelCreator(MPPMChannel, 2)
+    channel_3 = Instrument.ChannelCreator(MPPMChannel, 3)
+    channel_4 = Instrument.ChannelCreator(MPPMChannel, 4)
+
+    zero_all = Instrument.control(
+        "SENSe:CORRection:COLLect:ZERO:ALL?",
+        "SENSe:CORRection:COLLect:ZERO:ALL",
+        """Control the electrical offsets for all power meter channels."""
+    )
+
+    time = Instrument.control(
+        "SYSTem:TIME?", "SYSTem:TIME %g,%g,%g",
+        """Control the  the instrument’s internal time."""
+        # get_process=lambda v: tuple(float(x) for x in v.split(','))
+    )
+
+    error = Instrument.measurement(
+        "SYSTem:ERRor?", """Get the next error from the error queue."""
+    )
+
+    wavelength_all = Instrument.setting(
+        "SENSe:POWer:WAVelength:ALL %gNM",
+        """Control the sensor wavelength of this channel."""
```

Comment:
It is a `setting`, so the docstring should be "Set".
Also, it is named `_all` and the docstring states `this channel`.